src/Couchbase.Lite.Shared/Query/IndexConfiguration.cs

```diff
@@ -28,53 +31,48 @@ namespace Couchbase.Lite.Internal.Query
     /// </summary>
     public abstract class IndexConfiguration
     {
-        #region Properties
+        private const string Tag = nameof(IndexConfiguration);
 
         /// <summary>
         /// Gets the expressions to use to create the index
         /// </summary>
         public string[]? Expressions { get; }
 
-        internal C4QueryLanguage QueryLanguage { get; }
-
         internal C4IndexType IndexType { get; }
 
         internal abstract C4IndexOptions Options { get; }
 
-        #endregion
-
-        #region Constructor
-
-        internal IndexConfiguration(C4IndexType indexType, params string[] items)
-            : this(indexType, C4QueryLanguage.N1QLQuery)
-        {
-            Expressions = items;
-        }
-
-        internal IndexConfiguration(C4IndexType indexType, C4QueryLanguage queryLanguage)
+        internal IndexConfiguration(C4IndexType indexType, string[]? expressions)
         {
+            if(expressions != null) {
+                // Quick sanity check.  If I allow this to proceed, no error will happen but
+                // perhaps unintended behavior will.  The final string to pass will be empty
+                // and LiteCore will index the entire path element.  This is ok in some cases
+                // like ArrayIndexConfiguration, but it should be manifest with a null array
+                // and not an empty array or array of empty or array of null.
+                if(expressions.Length == 1 && expressions[0] == "") { 
```

Comment:
Yeah.... Good call.  I noticed something similar that I want to discuss this afternoon.